pkgs/stdenv/generic/builder.sh

```diff
@@ -4,18 +4,20 @@ for i in $initialPath; do
     PATH=$PATH${PATH:+:}$i/bin
 done
 
-mkdir $out
+mkdir "$out"
 
-echo "export SHELL=$shell" > $out/setup
-echo "initialPath=\"$initialPath\"" >> $out/setup
-echo "defaultNativeBuildInputs=\"$defaultNativeBuildInputs\"" >> $out/setup
-echo "defaultBuildInputs=\"$defaultBuildInputs\"" >> $out/setup
-echo "$preHook" >> $out/setup
-cat "$setup" >> $out/setup
+{
+    echo "export SHELL=$shell"
+    echo "initialPath=\"$initialPath\""
+    echo "defaultNativeBuildInputs=\"$defaultNativeBuildInputs\""
+    echo "defaultBuildInputs=\"$defaultBuildInputs\""
+    echo "$preHook"
+    cat "$setup"
+} > "$out/setup"
```

Comment:
This is a great change!  One of my pet peeves :)

The `>>` style is less readable and actually causes a ton of `open()` and `close()` syscalls